src/loaders/gltfloader.rs

```diff
@@ -0,0 +1,35 @@
+use gltf::mesh;
+
+use crate::geometry::Mesh;
+use crate::loaders::fileloader::FileLoader;
+
+pub struct GltfLoader {}
+
+impl FileLoader for GltfLoader {
+    fn load_file(filepath: &str) -> Vec<Mesh> {
+        let (document, buffers, _) = gltf::import(filepath).expect("Could not open Gltf file");
+
+        let mut ret: Vec<Mesh> = Vec::new();
+
+        for mesh in document.meshes() {
+            for primitive in mesh.primitives() {
```

Comment:
Is there a way to express this using flatMap? I read about it but ran into some trouble so just did it this way to get something compiling 😅

Reply:
Yes. You can do this with:
```rust
for primitive in document.meshes().flat_map(|mesh| mesh.primitives()) { ... }
```

Let me know if that doesn't work. I'll help you figure it out.

Reply:
:( there was lifetime issue where mesh and mesh.primitives() had the same lifetime so it didn't let me return it